### PR TITLE
Add Google user authentication

### PR DESCRIPTION
**Pull request by agentfll**
:hourglass: **Status**: _Ready to merge_
:tickets: **Ticket(s)**:

## :construction_worker: Changes

Google user authentication is implemented using token validation, so we don't need to actually ask the any server for any user information.

We currently issue bearer tokens with the user's ID in them, and validate them with a signature. If a user loses their token, they will be compromised forever under this scheme.

## :flashlight: Testing Instructions

1. Spin up a local `beluga` with `docker-compose up`
2. Point an instance of `fillet` at it, and log in.

### CODE REVIEW

**Comment by agentfll** [comment]
@chadlagore The tests were getting 403's when they where hitting `/events`--I disabled authentication in the test container as a fix. I'll add tests for `/sessions` before merging this.

File: tests/Dockerfile [hunk 1]
```diff
@@ -12,5 +12,6 @@ ADD . /app/
 WORKDIR /app/
 
 ENV BELUGA_TEST true
+ENV SECRET_BASE f5a9149cf77a47ddcc053a7731ec333a17b62238ff147345289a2d17fbd01eaec56946b5e339195484eb63140042e75c6ced01f46e196883eaee477240c46cdd
```
Comment (chadlagore) [inline tests/Dockerfile]: 🤔 is this a real secret? You can add env vars to our travis if you like.

File: tests/utils.py [hunk 2]
```diff
@@ -98,3 +98,37 @@ def wrapped_f(*args):
...
+
+    # Have to add the categoies before loading events
+    # due to foreign key constraint.
+    @add_db_categories([
+        {'category_id': 102, 'name': 'new_cat1'},
+        {'category_id': 101, 'name': 'new_cat2'},
+        {'category_id': 114, 'name': 'new_cat3'},
+        {'category_id': 117, 'name': 'new_cat4'}
+    ])
+    def add_events(self):
```
Comment (chadlagore) [inline tests/utils.py]: ~`add_events`~ `add_users` nice decorator use 💯  :+1:

File: tests/Dockerfile (hunk 1 above)
Comment (chadlagore) [reply tests/Dockerfile]: or just a dummy to get through the tests

Comment (agentfll) [reply tests/Dockerfile]: Just a dummy for tests, in real life it will generate one randomly at start

File: tests/utils.py (hunk 2 above)
Comment (agentfll) [reply tests/utils.py]: hahahaha whoops

**Pull request merged**